File: headapps/app/src/lib/factories/ford-escape-factory.ts
import { VehicleBuilder } from "../builders/vehicle-builder";
import { CarFactory } from "./car-factory";

export class FordEscapeFactory extends CarFactory {
  protected createBuilder(): VehicleBuilder {
    return new VehicleBuilder()
      .setColor("Red")
      .setBrand("Ford")
      .setModel("Escape");
  }
}

File: DesignPatterns/FactoryMethods/FordEscapeFactory.cs
using DesignPatterns.ModelBuilders;

namespace DesignPatterns.FactoryMethods
{
    public class FordEscapeFactory : CarFactory
    {
        public override string ModelKey => "Escape";

        protected override CarBuilder CreateBuilder()
        {
            return new CarBuilder()
                .setColor("Red")
                .setBrand("Ford")
                .setModel("Escape");
        }
    }
}
